File: VSTO - How to Customize Outlook's User Interface (Visual Studio 2012)/[VB]-VSTO - How to Customize Outlook's User Interface (Visual Studio 2012)/VB/uimanageroutlookaddin.addinutilities/iformregioncontrols.vb
' Copyright © Microsoft Corporation.  All Rights Reserved.
' This code released under the terms of the 
' Microsoft Public License (MS-PL, http://opensource.org/licenses/ms-pl.html.)


Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Runtime.InteropServices
Imports Outlook = Microsoft.Office.Interop.Outlook


' This interface is not CLS compliant because of its Office property.
<ComVisible(True)> _
<InterfaceType(ComInterfaceType.InterfaceIsDual)> _
<Guid("53ED8FA5-DBAD-40c4-8068-F20E7858DEB6")> _
<CLSCompliant(False)> _
Public Interface IFormRegionControls
    Property Inspector() As Outlook.Inspector
    Sub SetControlText(ByVal controlName As String, ByVal text As String)
End Interface

File: VSTO - How to Customize Outlook's User Interface (Visual Studio 2012)/[C#]-VSTO - How to Customize Outlook's User Interface (Visual Studio 2012)/C#/uimanageroutlookaddin.addinutilities/iformregioncontrols.cs
// Copyright © Microsoft Corporation.  All Rights Reserved.
// This code released under the terms of the 
// Microsoft Public License (MS-PL, http://opensource.org/licenses/ms-pl.html.)


using System;
using System.Collections.Generic;
using System.Text;
using System.Runtime.InteropServices;
using Outlook = Microsoft.Office.Interop.Outlook;


namespace UiManagerOutlookAddIn.AddinUtilities
{
    // This interface is not CLS compliant because of its Office property.
    [ComVisible(true)]
    [InterfaceType(ComInterfaceType.InterfaceIsDual)]
    [Guid("53ED8FA5-DBAD-40c4-8068-F20E7858DEB6")]
    [CLSCompliant(false)]
    public interface IFormRegionControls
    {
        Outlook.Inspector Inspector
        {
            get;
            set;
        }

        void SetControlText(String controlName, String text);
    }
}
